### PR TITLE
 Always raise ValueError on unknown variants

### LINKED ISSUE #122: Check if dataset generator exists and otherwise raise an exception
Currently, if the dataset generator receives an invalid dataset, it silently picks a default generator.
https://github.com/kiudee/cs-ranking/blob/5bdc0e414dde196761b7c827369b605a024bae42/csrank/dataset_reader/objectranking/object_ranking_data_generator.py#L35-L36
This is unexpected behavior and should be changed. If the dataset generator is unknown, an exception should be raised.

We should check all dataset generators/parsers for similar behavior.

### PR DESCRIPTION
**Pull request by timokau**
## Description

Fixes https://github.com/kiudee/cs-ranking/issues/122. I think I caught all instances, I verified this by grepping for ` not in `.

I kind of think most of those should either pass the function directly in the first place or use an enum instead. Either way, this change should be an improvement.

This is technically a breaking change, though practically not really important since it only changes behaviour in already broken code.

While working on this I noticed some surprising black behaviour (explained in the commit message), which I fixed in a separate commit.

## Motivation and Context

#122 

## How Has This Been Tested?

Ran the testsuite and pre-commit hooks.

## Does this close/impact existing issues?

Fixes #122.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by timokau** [comment]
It is best to review the commits individually, since one of them just adds a bunch of commas.

File: csrank/dataset_reader/choicefunctions/choice_data_generator.py [hunk 1]
```diff
@@ -15,7 +15,9 @@ def __init__(self, dataset_type="pareto", **kwargs):
             "pareto": self.make_globular_pareto_choices,
         }
         if dataset_type not in dataset_function_options.keys():
```
Comment (kiudee) [inline csrank/dataset_reader/choicefunctions/choice_data_generator.py:17]: Very minor nitpick at the original code: We sometimes use `obj not in dict.keys()` and sometimes `obj not in dict`.
Both are valid, but I think we should be consistent.
`obj not in dict` is in general regarded as the "more pythonic" syntax.

Comment (timokau) [reply csrank/dataset_reader/choicefunctions/choice_data_generator.py:17]: > Both are valid, but I think we should be consistent.

Agreed!

I kind of prefer the more explicit call to `.keys` for readability. It looks like you're right though that the direct `in` is the more "pythonic" way and also more efficient: https://stackoverflow.com/questions/1602934/check-if-a-given-key-already-exists-in-a-dictionary

File: csrank/tests/test_ranking.py [hunk 2]
```diff
@@ -63,8 +63,8 @@ def trivial_ranking_problem():
 
 def check_params_tunable(tunable_obj, params, rtol=1e-2, atol=1e-4):
     for key, value in params.items():
-        if key in tunable_obj.__dict__:
-            expected = tunable_obj.__dict__[key]
+        if hasattr(tunable_obj, key):
+            expected = getattr(tunable_obj, key)
```
Comment (kiudee) [inline csrank/tests/test_ranking.py:67]: Yes, good find. That’s definitely the preferred way to check for attributes

**Pull request merged**